tests/common/mod.rs

```diff
@@ -0,0 +1,132 @@
+extern crate fs_extra;
+extern crate lal;
+extern crate loggerv;
+extern crate tempdir;
+extern crate walkdir;
+
+use common::{
+    fs_extra::dir::{copy, CopyOptions},
+    loggerv::init_with_verbosity,
+    tempdir::TempDir,
+};
+
+use std::{
+    env,
+    path::{Path, PathBuf},
+    sync::Once,
+};
+
+use lal::{BackendConfiguration, Config, LocalBackend};
+
+pub mod build_and_stash_update_self;
+pub mod change_envs;
+pub mod check_propagation;
+pub mod clean_check;
+pub mod export_check;
+pub mod fetch_release_build_and_publish;
+pub mod get_environment_and_fetch;
+pub mod has_config_and_manifest;
+pub mod init_force;
+pub mod list_everything;
+pub mod no_publish_non_release_builds;
+pub mod query_check;
+pub mod remove_dependencies;
+pub mod run_scripts;
+pub mod shell_echo;
+pub mod shell_permissions;
+pub mod status_on_experimentals;
+pub mod update_save;
+pub mod verify_checks;
+
+pub use build_and_stash_update_self::build_and_stash_update_self;
+pub use change_envs::change_envs;
+pub use check_propagation::check_propagation;
+pub use clean_check::clean_check;
+pub use export_check::export_check;
+pub use fetch_release_build_and_publish::fetch_release_build_and_publish;
+pub use get_environment_and_fetch::get_environment_and_fetch;
+pub use has_config_and_manifest::has_config_and_manifest;
+pub use init_force::init_force;
+pub use list_everything::list_everything;
+pub use no_publish_non_release_builds::no_publish_non_release_builds;
+pub use query_check::query_check;
+pub use remove_dependencies::remove_dependencies;
+pub use run_scripts::run_scripts;
+pub use shell_echo::shell_echo;
+pub use shell_permissions::shell_permissions;
+pub use status_on_experimentals::status_on_experimentals;
+pub use update_save::update_save;
+pub use verify_checks::verify_checks;
+
+pub struct TestState {
+    pub backend: LocalBackend,
+    pub testdir: PathBuf,
+
+    // Keep the tempdir with TestState.
+    // The directory will be cleaned with the TestState is `Drop`ed.
+    pub tempdir: TempDir,
+}
+
+static START: Once = Once::new();
+
+pub fn setup() -> TestState {
+    START.call_once(|| {
+        env::set_var("SSL_CERT_FILE", "/etc/ssl/certs/ca-certificates.crt");
+        env::set_var("LAL_CODE_DIR", env::current_dir().unwrap().as_os_str());
+
+        // print debug output and greater from lal during tests
+        init_with_verbosity(2).expect("Setting up test logging");
+    });
+
+    let mut demo_config = PathBuf::from(env::var("LAL_CODE_DIR").unwrap());
+    demo_config.push("./configs/demo.json");
+
+    let mut testdir = PathBuf::from(env::var("LAL_CODE_DIR").unwrap());
+    testdir.push("tests");
```

Comment:
we set the evar to unwrap read from it again? are we even picking up on the evar from tests anyway?

Reply:
Yes! Since I've removed all calls to `set_current_dir`, keeping a reference to the code isn't necessary any more.